### PR TITLE
service: migrate schema to new IdentifierSchema api

### PR DESCRIPTION
**Pull request by ppanero**
- requires https://github.com/inveniosoftware/marshmallow-utils/pull/40
- requires https://github.com/inveniosoftware/invenio-vocabularies/pull/71
- closes https://github.com/inveniosoftware/invenio-rdm-records/issues/535

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -170,8 +170,7 @@ def full_record(users):
             "version": "v1.0",
             "rights": [{
                 "title": "Creative Commons Attribution 4.0 International",
-                "scheme": "spdx",
-                "identifier": "cc-by-4.0",
+                "id": "cc-by-4.0",
```
Comment (ppanero) [inline tests/conftest.py:173]: @lnielsen revisit this when tackling vocabularies... tests/serializers need to be checked

**Review by fenekku — APPROVED** [review]
Just need to wait on the other one to get merged and this to bump dependencies.

File: invenio_rdm_records/services/schemas/__init__.py [hunk 2]
```diff
@@ -23,6 +25,10 @@
 from .pids import PIDSchema
 from .versions import VersionsSchema
 
+record_pids_schemes = LocalProxy(
```
[the remark this answers is not in the document]
Comment (ppanero) [reply invenio_rdm_records/services/schemas/__init__.py:28]: You are right... (tested, tests dont fail). I was following this <https://github.com/inveniosoftware/invenio-oauth2server/blob/f74bbe1ea19656831ac94946999100b65e6fe7dd/invenio_oauth2server/provider.py#L25>... Also I recall @lnielsen saying we should use a `LocalProxy`... but you are right there might not be a need for it...

File: invenio_rdm_records/services/schemas/metadata.py [hunk 3]
```diff
@@ -12,15 +12,42 @@
...
+
+record_references_schemes = LocalProxy(
+    lambda: current_app.config["RDM_RECORDS_REFERENCES_SCHEMES"]
+)
+
+
+record_location_schemes = LocalProxy(
+    lambda: current_app.config["RDM_RECORDS_LOCATION_SCHEMES"]
+)
+
```
[the remark this answers is not in the document]
Comment (ppanero) [reply invenio_rdm_records/services/schemas/metadata.py:50]: Lets tackle in comment above.

File: invenio_rdm_records/services/schemas/metadata.py [hunk 4]
```diff
@@ -210,14 +243,9 @@ def _is_uri(uri):
         return False
 
 
-class RightsSchema(IdentifierSchema):
+class RightsSchema(Schema):
```
[the remark this answers is not in the document]
Comment (fenekku) [reply invenio_rdm_records/services/schemas/metadata.py:246]: ok...

**Comment by ppanero** [comment]
Require invenio-vocabularies released before passing tests, but ✅ in local.

Did a full instance install with the required modules from local and created a *full record* (i.e. all fields from the deposit form). All working correctly.

**Pull request merged**